src/main/kotlin/no/nav/syfo/behandler/BehandlerService.kt

```diff
@@ -101,14 +96,21 @@ class BehandlerService(
     }
 
     private fun getBehandlerDialogmelding(behandler: Behandler): PBehandlerDialogmelding? {
-        if (behandler.personident != null) {
-            return database.getBehandlerDialogmeldingMedPersonIdentForPartnerId(
+        return when {
+            behandler.personident != null -> database.getBehandlerDialogmeldingMedPersonIdentForPartnerId(
                 behandlerPersonIdent = behandler.personident,
                 partnerId = behandler.partnerId,
             )
+            behandler.hprId != null -> database.getBehandlerDialogmeldingMedHprIdForPartnerId(
+                hprId = behandler.hprId,
+                partnerId = behandler.partnerId,
+            )
+            behandler.herId != null -> database.getBehandlerDialogmeldingMedHerIdForPartnerId(
+                herId = behandler.herId,
+                partnerId = behandler.partnerId,
+            )
```

Comment:
Hva om behandler kun har hpr første gang, så lagrer vi det i databasen; og neste gang får vi også med fnr? Får vi da treff på den første sjekken her, men ikke treff i databasen?

Reply:
Hmja, det har du nok rett i. Sånn det er nå vil man vel da ende opp med å lage en "ny" behandler (bare med fnr i tillegg) i behandler-tabellen og en ny kobling til arbeidstakeren.
Vi kan jo ta hensyn til det her ved å gjøre flere sjekker og databasekall når man ikke får treff. I tillegg bør vi kanskje også oppdatere behandleren i databasen (dersom vi får treff) slik at den får fnr (og potensielt andre felter) oppdatert.

Hva tenker du/dere? Det baller jo litt på seg dette her 😬

Reply:
Man får jo returnert riktig behandler nå, så da kan det kanskje være. Det kan evt. være en egen oppgave etter denne?
Det er vel bra nok om man får en behandler med partnerid, så det ikke gjør noe om vi har flere innslag av samme behandler 🤷

Reply:
Ja, men vi bruker vel en del av feltene vi har lagret for behandler til sendingen av dialogmelding, så sikkert ikke dumt om de er oppdatert. Men enig i at det nok bør gjøres i en egen oppgave ja :)